File: data-structures-and-algorithms-ashrafi/java/src/recursive/problem14/Problem14.java
public class Problem14 {
    
    public static void printHanoi(int n, char S, char D, char A){
        if (n == 1) {
            System.out.println(S + " -> " + A);
            System.out.println(A + " -> " + D);
        } else {
            printHanoi(n - 1, S, D, A);
            System.out.println(S + " -> " + A);
            printHanoi(n - 1, D, S, A);
            System.out.println(A + " -> " + D);
            printHanoi(n - 1, S, D, A);
        }
    }

    public static void main(String[] args) {
        int n = 2;
        char S = 'S';
        char D = 'D';
        char A = 'A';

        printHanoi(n, S, D, A);
    }
}

File: data-structures-and-algorithms-ashrafi/python/src/recursive/problem_14/problem_14.py
def printHanoi(n, S, D, A):
    if n == 1:
        print(f"{S} -> {A}")
        print(f"{A} -> {D}")
    else:
        printHanoi(n - 1, S, D, A)
        print(f"{S} -> {A}")
        printHanoi(n - 1, D, S, A)
        print(f"{A} -> {D}")
        printHanoi(n - 1, S, D, A)

def main():
    n = 2
    charS = 'S'
    charD = 'D'
    charA = 'A'
    
    printHanoi(n, charS, charD, charA)


if __name__ == "__main__":
    main()